pkg/controller/migmigration/task.go

```diff
@@ -89,139 +99,201 @@ type Itinerary struct {
-		{phase: EnsureStageRestore, any: HasPVs | HasISs},
-		{phase: StageRestoreCreated, any: HasPVs | HasISs},
-		{phase: EnsureStagePodsDeleted, all: HasStagePods},
-		{phase: EnsureStagePodsTerminated, all: HasStagePods},
-		{phase: EnsureAnnotationsDeleted, any: HasPVs | HasISs},
-		{phase: EnsureInitialBackupReplicated},
-		{phase: PostBackupHooks},
-		{phase: PreRestoreHooks},
-		{phase: EnsureFinalRestore},
-		{phase: FinalRestoreCreated},
-		{phase: PostRestoreHooks},
-		{phase: Verification, all: HasVerify},
-		{phase: Completed},
+		{
+			Name: StepPrepare,
+			Phases: []Phase{
+				{Name: Created},
+				{Name: Started},
+				{Name: StartRefresh},
+				{Name: WaitForRefresh},
+				{Name: Prepare},
+				{Name: EnsureCloudSecretPropagated},
+			},
+		},
+		{
+			Name: StepBackup,
+			Phases: []Phase{
+				{Name: PreBackupHooks},
+				{Name: EnsureInitialBackup},
+				{Name: InitialBackupCreated},
+			},
+		},
+		{
+			Name: StepStageBackup,
+			Phases: []Phase{
+				{Name: EnsureStagePodsFromRunning, all: HasPVs},
+				{Name: EnsureStagePodsFromTemplates, all: HasPVs},
+				{Name: EnsureStagePodsFromOrphanedPVCs, all: HasPVs},
+				{Name: StagePodsCreated, all: HasStagePods},
+				{Name: AnnotateResources, any: HasPVs | HasISs},
+				{Name: RestartRestic, all: HasStagePods},
+				{Name: ResticRestarted, all: HasStagePods},
+				{Name: QuiesceApplications, all: Quiesce},
+				{Name: EnsureQuiesced, all: Quiesce},
+				{Name: EnsureStageBackup, any: HasPVs | HasISs},
+				{Name: StageBackupCreated, any: HasPVs | HasISs},
+				{Name: EnsureStageBackupReplicated, any: HasPVs | HasISs},
+			},
+		},
+		{
+			Name: StepStageRestore,
+			Phases: []Phase{
+				{Name: EnsureStageRestore, any: HasPVs | HasISs},
+				{Name: StageRestoreCreated, any: HasPVs | HasISs},
+				{Name: EnsureStagePodsDeleted, all: HasStagePods},
+				{Name: EnsureStagePodsTerminated, all: HasStagePods},
+			},
+		},
+		{
+			Name: StepRestore,
+			Phases: []Phase{
+				{Name: EnsureAnnotationsDeleted, any: HasPVs | HasISs},
+				{Name: EnsureInitialBackupReplicated},
+				{Name: PostBackupHooks},
+				{Name: PreRestoreHooks},
+				{Name: EnsureFinalRestore},
+				{Name: FinalRestoreCreated},
+				{Name: PostRestoreHooks},
+			},
+		},
+		{
+			Name: StepFinal,
+			Phases: []Phase{
+				{Name: Verification, all: HasVerify},
+				{Name: Completed},
+			},
+		},
 	},
 }
 
 var CancelItinerary = Itinerary{
 	Name: "Cancel",
 	Steps: []Step{
-		{phase: Canceling},
-		{phase: DeleteBackups},
-		{phase: DeleteRestores},
-		{phase: EnsureStagePodsDeleted, all: HasStagePods},
-		{phase: EnsureAnnotationsDeleted, any: HasPVs | HasISs},
-		{phase: Canceled},
-		{phase: Completed},
+		{
+			Name: StepFinal,
+			Phases: []Phase{
+				{Name: Canceling},
+				{Name: DeleteBackups},
+				{Name: DeleteRestores},
+				{Name: EnsureStagePodsDeleted, all: HasStagePods},
+				{Name: EnsureAnnotationsDeleted, any: HasPVs | HasISs},
+				{Name: DeleteMigrated},
```

Comment:
@pranavgaikwad we've recently trimmed a few steps from the `CancelItinerary` with our rollback on demand changes. See the diff, we should match the phase list from master.

Reply:
Thank you for pointing that. I missed this in a rebase